tests/scan.py

```diff
@@ -0,0 +1,31 @@
+# Copyright 2020 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Unittests for scan method."""
+
+import unittest
+
+import jax.numpy as jn
+
+import objax
+
+
+class TestScan(unittest.TestCase):
+    def test_scan(self):
+        def cell(carry, x):
+            return jn.array([2]) * carry * x, jn.array([2]) * carry * x
+
+        a, b = objax.functional.scan(cell, jn.zeros((2,)), jn.zeros((3,)))
```

Comment:
The test is a bit too light, like using zeros to test a multiplication leads to more zeros. Also you would not find out if the carry is properly as the first argument or not.
1. Use non-zero inputs
2. Make carry and output different values (right now they're the same)
3. Test return values (testing the values will test their shape as well).

Reply:
Change requested

Reply:
1. Done
2. Done
2. Done

Reply:
I don't think 2 is what I had in mind, I wanted to return something different.

Right now it returns 
```
return jn.array([2]) * carry * x, jn.array([2]) * carry * x
```
Which are both the same, so it could return carry as output and the test wouldn't catch it.

What I in mind would be something like:
```
return jn.array([2]) * carry * x, jn.array([3]) * carry * x
```

Reply:
Thank you for clarifying and providing the suggestion. I've updated the code to do as suggested